src/test/java/com/transferwise/common/gaffer/test/complextest1/ComplexTest.java

```diff
@@ -206,6 +207,26 @@ public void testFullCommitWithHibernate() {
     Assert.assertEquals(1, getTableCount(TABLE_ACCOUNTS));
   }
 
+  @Test
+  public void testConnReleasedAfterTxRollback() {
+    setUp();
+    try {
+      getConfig().setFailPasswordCreation(true);
+      Stat stat = new Stat("testConnReleasedAfterTxRollback");
+      try {
+        getClientsService().deleteClient(CLIENT_NAME);
+      } finally {
+        log.info("{}", stat);
+      }
+      Assert.fail("Test has invalid logic, recheck it!");
+    } catch (Exception e) {
+      log.debug(e.getMessage(), e);
+    }
+
+    Assert.assertEquals(2, getClientDataSource().getAllConnectionGetsCount());
+    // somehow make sure connections are closed
```

Comment:
didn't find a good way to check if associated connections have actually been closed